### PR TITLE
backup: add version checker for store based backup

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve?

Issue Number: ref #52534

Problem Summary:
Due to https://github.com/tikv/tikv/issues/17168, the current tikv does not support backup with subranges very well. 
So we need a version checker in case of use store based backup running on old version of tikv.(There is no correctness issue, but the performance may not work as expected.)

### What changed and how does it work?
Add a version check
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/version/version.go [hunk 1]
```diff
@@ -233,6 +233,15 @@ func CheckVersionForBR(s *metapb.Store, tikvVersion *semver.Version) error {
 			s.Address, tikvVersion)
 	}
 
+	// 8.2 br(store based backup) does not support tikv <= 8.1
+	// due to the performance issue https://github.com/tikv/tikv/issues/17168
+	// TODO: we can remove this check if the performance issue is fixed and cherry-pick
+	if (BRVersion.Major > 8 || (BRVersion.Major == 8 && BRVersion.Minor >= 2)) &&
```
Comment (Leavrth) [inline br/pkg/version/version.go:239]: This check is always true?

**Review by Leavrth — APPROVED** [review]

File: br/pkg/version/version.go (hunk 1 above)
Comment (YuJuncen) [reply br/pkg/version/version.go:239]: If this PR wasn't cherry-picked, it looks always true. But I guess this won't be really harmful.

**Review by YuJuncen — APPROVED** [review]
LGTM

**Pull request merged**